rpc/client.go

```diff
@@ -186,6 +186,25 @@ func DialContext(ctx context.Context, rawurl string) (*Client, error) {
 	}
 }
 
+func DialWithAuth(ctx context.Context, rawurl string, auth HeaderAuthProvider) (*Client, error) {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return nil, err
+	}
+	switch u.Scheme {
+	case "http", "https":
```

Comment:
Returning an error here or in the HTTPProviderWithAuth if the authorprovider is nil might be nice. You'll get an error when doing the connection, but erring here makes it clear what the problem is.

(Also needs to be mirrored for WS)

Reply:
`DialHTTPWithAuth` and `DialWebsocketWithAuth` now check if the `auth` is not nil, and I added test cases for it.